### PR TITLE
Mse sc latest snapshot

### PR DESCRIPTION
**Pull request by simonc-613**
Adding the ability to specify a snapshot label to restore from the latest snapshot

Use-case:
Mongo restore test nodes need to find and mount the latest snapshot and test the data so cannot use a specific snapshot id easily

### CODE REVIEW

File: lib/puppet/type/ec2_volume.rb [hunk 1]
```diff
@@ -71,6 +71,13 @@
     end
   end
 
+  newproperty(:snapshot_label) do
+    desc 'Label of the snapshot that this volume should be created from'
+    validate do |value|
+      fail 'snapshot_label should be an string' unless value.is_a?(String)
```
Comment (chrisjomaron) [inline lib/puppet/type/ec2_volume.rb]: 'a' string

**Comment by simonc-613** [comment]
@chrisjomaron fixed it, copy and paste fail so fixed all occurrences.

Please could you merge and tag with the next version? I think it is `2.0.0g`

File: lib/puppet/provider/ec2_volume/v2.rb [hunk 2]
```diff
@@ -78,11 +78,32 @@ def exists?
...
-    snapshot = resource[:snapshot_id] ? resource[:snapshot_id] : false
+    snapshot = latest_snapshot
     config['snapshot_id'] = snapshot if snapshot
     config
   end
 
+  def latest_snapshot
+    return resource[:snapshot_id] if resource[:snapshot_id]
+    return false unless resource[:snapshot_label]
+    Puppet.notice("Attempting to fetch snapshot ID")
```
Comment (dstockton) [inline lib/puppet/provider/ec2_volume/v2.rb]: Include label

File: lib/puppet/provider/ec2_volume/v2.rb [hunk 3]
```diff
@@ -78,11 +78,32 @@ def exists?
...
+
+  def find_snapshots
+    filters = [{
+      name: 'description',
+      values: [resource[:snapshot_label]]
+    }, {
+      name: 'status',
+      values: ['completed']
+    }]
+    ec2.describe_snapshots(filters: filters).snapshots
```
Comment (dstockton) [inline lib/puppet/provider/ec2_volume/v2.rb]: wrap with retries

Comment (simonc-613) [reply lib/puppet/provider/ec2_volume/v2.rb]: @dstockton not been ignoring you, honest!
Took a look through this and is this not already wrapped with retries from within `def create` in [L136](https://github.com/simonc-613/puppetlabs-aws/blob/master/lib/puppet/provider/ec2_volume/v2.rb#L136)?

Comment (dstockton) [reply lib/puppet/provider/ec2_volume/v2.rb]: I guess yes but you're changing the profile of how we retry...
e.g. any 1 of several API call failures results in a retry (up to max of 10 attempts) vs any single API call failure results in a retry of that specific API call (up to a max of 10 attempts)

Happy to just let it be - it's already in use so we can always re-visit if it becomes an issue

Comment (simonc-613) [reply lib/puppet/provider/ec2_volume/v2.rb]: No worries, you asked for a text change so I can add it in, was just looking at the other code and was wondering

Comment (simonc-613) [reply lib/puppet/provider/ec2_volume/v2.rb]: @dstockton see #11

**Pull request merged**